bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productlist/ProductListImpl.java

```diff
@@ -207,6 +247,11 @@ public Collection<ProductListItem> getProducts() {
         }
     }
 
```

Comment:
Does this support inserting a XF to a certain page of the products or an XF will be added to all pages?

(My understanding was that we always display the page size number of products (to not mix with graphql pagination) and in addition we show the XF among the products at the position specified by the author to occupy a certain row of a certain page.)

Reply:
no. we show all the products, then add all the xf banners which span an entire row. the banners will be positioned between rows of products using CSS

Reply:
Iiuc banners will not be always shown on the first page? In the grooming we talked about still positioning them "before product XYZ". Why isn't this implemented?

Having the banner only on the first page, will be too limiting. (remember the use case with the advisor bot banner being shown 2nd row on the 2nd page?)

Reply:
> Iiuc banners will only be shown on the first page? In the grooming we talked about still positioning them "before product XYZ". Why isn't this implemented?

In the grooming we discussed about trying that but there is no way to do it. It is dependent on the grid size. if there even is a grid. a customer can choose to use another layout system.

And, since the the banner spans an entire row, it makes more sense for it to be positioned on a specific row. 
again, because we don't know what layout will be used, we don't provide a CSS style for this.
Rather we leave it to the customer.

Reply:
Now that the CSS class is configured specified in the policy, couldn't we configure the page it should be shown there as well? 

2nd Page, 2nd Row: 
- `fragmentCssClass = 2nd-row`
- `fragmentPage = 2`